src/CloudNative.CloudEvents/Extensions/Sampling.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright 2021 Cloud Native Foundation. 
+// Licensed under the Apache 2.0 license.
+// See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace CloudNative.CloudEvents.Extensions
+{
+    public static class Sampling
```

Comment:
nice - I like the approach of using extension methods for these.

Reply:
Thanks. I think we'll want to document this pattern carefully once we're really happy with it. There may well be some subtleties to think about in order to make it consumer-friendly when handling a CloudEvent created without the extensions known beforehand.